Core/src/main/java/com/plotsquared/core/plot/BlockBucket.java

```diff
@@ -110,54 +110,62 @@ public void compile() {
         if (isCompiled()) {
             return;
         }
-        this.compiled = true;
-        String string = this.input.toString();
-        if (string.isEmpty()) {
-            this.single = null;
-            this.pattern = null;
-            return;
-        }
-        // Convert legacy format
-        boolean legacy = false;
-        String[] blocksStr = string.split(",(?![^\\(\\[]*[\\]\\)])");
-        if (blocksStr.length == 1) {
-            try {
-                Matcher matcher = regex.matcher(string);
+        // Synchronized as BlockBuckets may require compilation asynchronously due to async chunk generation on Paper servers
+        synchronized (this) {
+            if (isCompiled()) {
+                return;
+            }
+            String string = this.input.toString();
+            if (string.isEmpty()) {
+                this.single = null;
+                this.pattern = null;
+                this.compiled = true;
+                return;
+            }
+            // Convert legacy format
+            boolean legacy = false;
+            String[] blocksStr = string.split(",(?![^\\(\\[]*[\\]\\)])");
```

Comment:
According to regex101 the inverted slashes are obsolete, the pattern matches still.

Reply:
They're there to avoid errors in java